components/skus/browser/skus_service_impl.cc

```diff
@@ -140,10 +158,28 @@ void SkusServiceImpl::CredentialSummary(
                                              std::move(cbs), domain);
 }
 
+void SkusServiceImpl::OnFetchOrderCredentials(
+    const std::string& domain,
+    const std::string& order_id,
+    skus::mojom::SkusService::FetchOrderCredentialsCallback callback,
+    const std::string& result) {
+  for (const auto& obs : observers_) {
+    obs->OnFetchOrderCredentials();
+  }
+
+  if (callback) {
+    std::move(callback).Run(result);
+  }
+}
+
 void SkusServiceImpl::OnCredentialSummary(
     const std::string& domain,
     mojom::SkusService::CredentialSummaryCallback callback,
     const std::string& summary_string) {
+  for (const auto& obs : observers_) {
```

Comment:
Curious we should have `callback` and oboserver both for this one event?

Reply:
Discussed in DM - but wanted to share the findings for other reviewers 😄 

I think we need both because this can be called from JavaScript (basically `window.chrome.braveSkus.credential_summary`). In that case, it's async and will likely have a `then` handler. That callback is provided in the V8 level automatically when someone is visiting the account.brave.com website and clicking `Refresh Brave VPN`. Here's what happens for that case:
- Page calls JS SKU SDK (the WASM compiled one)
- if `window.chrome.braveSkus.credential_summary` is present, it will call that. Since being on account.brave.com satisfies that criteria it will be present and the JS SKU SDK will make the call
- V8 bindings intercept `window.chrome.braveSkus.credential_summary` and have a context, which includes the callback.
- After the method finishes executing, the V8 layer can execute the callback (basically resolve the promise)

For that reason, I don't think we can remove the callback (it needs to work back to V8 level)

The observer is needed for a use-case where we are unable to provide a callback. Using the above use-case (user clicking the `Refresh Brave VPN` button), Android needs a way to be notified when this happens. When that happens, we are able to immediately begin showing the native VPN interface on Android (because it is like the purchase happened)

Reply:
Typing that out, I am already seeing logic being placed here which accomplishes what we want to do with the observer. I think we can remove the observer entirely

Will chat more with @deeppandya in DM 👍

Reply:
Code removed 👍